modules/openglqt/src/glslsyntaxhighlight.cpp

```diff
@@ -0,0 +1,318 @@
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/openglqt/glslsyntaxhighlight.h>
+#include <modules/qtwidgets/inviwoqtutils.h>
+
+#include <warn/push>
+#include <warn/ignore/all>
+#include <QFontDatabase>
+#include <warn/pop>
+
+#include <array>
+#include <string_view>
+
+using namespace std::literals;
+
+namespace inviwo {
+
+GLSLSyntaxHighlight::GLSLSyntaxHighlight()
+    : Settings("GLSL Syntax Highlighting")
+    , font("font", "Font", utilqt::getMonoSpaceFonts(), utilqt::getDefaultFontIndex())
+    , fontSize("fontSize", "Size", syntax::fontSize, 1, 72)
+
+    , textColor("text", "Text", util::ordinalColor(syntax::text))
+    , backgroundColor("background", "Background", util::ordinalColor(syntax::background))
+    , highLightColor("highLight", "HighLight", util::ordinalColor(syntax::highLight))
+    , keywordColor("keyword", "Keyword", util::ordinalColor(syntax::keyword))
+    , builtinVarColor("bultinVar", "Built-in Variable", util::ordinalColor(syntax::builtinVars))
+    , typeColor("type", "Type", util::ordinalColor(syntax::type))
+    , builtinFuncColor("bultinFunc", "Built-in Function", util::ordinalColor(syntax::builtinFuncs))
+    , commentColor("comment", "Comment", util::ordinalColor(syntax::comment))
+    , preProcessorColor("preProcessor", "Pre-Processor", util::ordinalColor(syntax::preProcessor))
+    , litteralColor("literal", "String Literal", util::ordinalColor(syntax::literal))
+    , constantColor("constants", "Constant", util::ordinalColor(syntax::constant))
+    , mainColor("main", "Main Function", util::ordinalColor(syntax::main)) {
+
+    addProperties(font, fontSize, textColor, backgroundColor, highLightColor, commentColor,
+                  typeColor, keywordColor, builtinVarColor, builtinFuncColor, preProcessorColor,
+                  litteralColor, constantColor, mainColor);
+
+    load();
+}
+
+namespace {
+
+// define GLSL types and keywords using regular expressions. \b indicates word boundaries.
+constexpr std::array types = {"float"sv,         "[bi]?vec[2-4]"sv, "int"sv,
```

Comment:
Leaving a reference for where these syntax/variables were retrieved from would be good in case one wants to update them.

Reply:
I have no idea, I just moved them.

Reply:
`double` and `dvec*` should be included here, I think.
see also https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)